File: Interview Preparation Kit - CPP/03. Dictionaries and Hashmaps/001. Hash Tables - Ransom Note.cpp
// Problem: https://www.hackerrank.com/challenges/ctci-ransom-note/problem
// Score: 25

#include <iostream>
#include <string>
#include <map>
using namespace std;


int main(){
    int m, n;
    cin >> m >> n;

    map<string, int> magazine;

    for (int i = 0; i < m; i++) {
        string word;
        cin >> word;

        if (magazine.find(word) == magazine.end()){
            magazine[word] = 1;
        }
        else {
            magazine[word]++;
        }
    }


    bool ans = true;
    for (int i = 0; i < n; i++){
        string word;
        cin >> word;

        if (magazine.find(word) == magazine.end() || magazine[word] == 0){
            ans = false;
            break;
        }
        else {
            magazine[word]--;
        }
    }


    if (ans) cout << "Yes" << endl;
    else cout << "No" << endl;

    return 0;
}

File: Interview Preparation Kit - Python/03. Dictionaries and Hashmaps/001. Hash Tables - Ransom Note.py
# Problem: https://www.hackerrank.com/challenges/ctci-ransom-note/problem
#  Score: 25


from collections import Counter


def checkMagazine(magazine, note):
    if Counter(note) - Counter(magazine) == {}:
        return 'Yes'
    else:
        return 'No'


m, n = map(int, input().split())
magazine = list(input().split())
note = list(input().split())
print(checkMagazine(magazine, note))
